sig-testing/flakiness-sla.md

```diff
@@ -0,0 +1,40 @@
+# Kubernetes Testing Flakiness SLA
+
+This document captures the expectations of the community about flakiness in
+our tests and our test infrastructure. It sets out an SLA (Service Level 
+Agreement) for flakiness in our tests, as well as actions that we will
+take when we are out of SLA.
+
+
+## Definition of a "Flake"
+
+We'll start by the definition of a _flake_. A test result is considered to
+be a flake if it both fails and passes without any changes in the code
+or environment being tested. Changes in the environment are
+notably hard to measure or control for, but we'll do our best. Changes in
+code are obviously quite easy to control.
+
+## Flakiness SLA
+We will measure flakiness based on pull requests that are run through e2e
+testing. The metric that we will monitor is: Flakes per PR. This metric
+will be calculated on a daily and weekly basis using:
+
+Sum(Flakes) / Total(PRs)
+
+Our current SLA is that this metric will be less than 0.01
+(1% of all PRs have flakes).
+
+## Activities on SLA violation
+When the project is out of SLA for flakiness, we will stop the merge queue.
+Only PRs with the label "flake-fix" will be allowed to merge. The merge queue
+will remain blocked until a sufficient number of flake-fixes have been
+submitted to restore SLA.
+
+## Non-goals: Flakiness at HEAD
+We will _not_ currently measure flakiness for continuous e2e that
+are run at HEAD.
```

Comment:
See above, we can't auto-detect flakes, but we can auto-detect the human response to flakes.

We have per-test flakiness at HEAD, we could use that too, but I like the PR driven metric better because I believe it will be more stable (and as I mention we already block the queue on HEAD e2e failures in many cases)